frame/vamm/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+#[frame_support::pallet]
+pub mod pallet {
+	// ----------------------------------------------------------------------------------------------------
+	//                                       Imports and Dependencies
+	// ----------------------------------------------------------------------------------------------------
+
+	use frame_support::pallet_prelude::*;
+	use frame_system::{ensure_signed, pallet_prelude::*};
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                    Declaration Of The Pallet Type
+	// ----------------------------------------------------------------------------------------------------
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                             Config Trait
+	// ----------------------------------------------------------------------------------------------------
+
+	// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		#[allow(missing_docs)]
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+	}
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                             Pallet Types
+	// ----------------------------------------------------------------------------------------------------
+
+	// ----------------------------------------------------------------------------------------------------
+	//                                            Runtime Events
+	// ----------------------------------------------------------------------------------------------------
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
```

Comment:
Doesn't `clippy` complain about `deposit_event` being unused?

Reply:
It does. 
but since I was running `pre-commit` and it didn't show the error I thought it was okay. Running `cargo clippy` as a stand alone command does show errors. 
Fixed, thanks for spotting this!